### PR TITLE
Create shell model and shell file

### PR DESCRIPTION
**Pull request by narensrini**

### CODE REVIEW

File: shells/shells.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+func getLastLineFish() string {
+	var fname = os.Getenv("HOME") + "/.config/fish/fish_history"
+	file, err := os.Open(fname)
+
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
```
Comment (rossem) [inline shells/shells.go]: can we move opening the file into a separate function? and just do something like

`var scanner = reverse.NewScanner(getFile())`

File: shells/shells.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+	scanner := reverse.NewScanner(file)
+	scanner.Scan()
+	return scanner.Text()
+}
+
+func getLastLineFish() string {
+	var fname = os.Getenv("HOME") + "/.config/fish/fish_history"
```
Comment (rossem) [inline shells/shells.go]: make path a constant

File: shells/shells.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	file, err := os.Open(fname)
+
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	scanner := reverse.NewScanner(file)
```
Comment (rossem) [inline shells/shells.go]: thought we agreed on using `var` for single assignments

File: shells/shells.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+var CurrentShell models.Shell
+
+func init() {
+	CurrentShell = models.Shell{Name: "fish", GetLastCmd: getLastLineFish}
+}
+
+func getLastLineBash() string {
+	var fname = os.Getenv("HOME") + "/.bash_history"
```
Comment (rossem) [inline shells/shells.go]: path can be a constant as well

File: shells/shells.go (hunk 2 above)
Comment (narensrini) [inline shells/shells.go]: According to the spec, you can't have a function call (`os.Getenv`) for constants as they're evaluated during compile time.

https://golang.org/doc/effective_go.html#constants

Comment (rossem) [inline shells/shells.go]: I mean making `"/.config/fish/fish_history"` a constant :)

Comment (rossem) [inline shells/shells.go]: also same thing for `/.bash_history`, and we can make a function to have a cleaner abstraction for this, something like:

``` go
func getPathFromHome(path string) string {
    return os.Getenv("HOME") + path
}
```

**Comment by rossem** [comment]
:bear:

**Pull request merged**